### PR TITLE
Add dropdown menu for Popup/ListPopup and TextInputLayout Filled/Outlined

### LINKED ISSUE #43: Dropdown component
**Decathlon.design link**

https://www.decathlon.design/726f8c765/p/42d824-dropdown-beta

**Material.io link**

https://material.io/components/menus

### PR DESCRIPTION
**Pull request by GerardPaligot**
## Changes description 🧑‍💻

* Upgrade Kotlin to 1.5.31.
* Add vtmnContentTertiary Vitamin semantic color.
* Create internal dropdown module for menus and text inputs.
* Create menus module for dropdown popup and list popup.
* Add dropdown for TextInput Filled/Outlined.

## Context 🤔

Closes #43 

## Checklist ✅

 - [x] I have reviewed the submitted code
 - [x] I have tested on a phone device/emulator
 - [x] I have tested on a tablet device/emulator
 - [x] I have tested on a kiosk device/emulator

#### Phone

https://user-images.githubusercontent.com/1913901/140509406-c3f8c580-9280-4679-aad7-87da7dc08aa0.mp4

#### Tablet

https://user-images.githubusercontent.com/1913901/140509519-8b687b1b-fd5d-4eb7-83f9-d419b13be082.mp4

#### Kiosk

https://user-images.githubusercontent.com/1913901/140509460-94178ad9-3b7a-4bd6-b84b-aa34930c7c1f.mp4

### CODE REVIEW

**Review by MaximeJineMaraval — CHANGES_REQUESTED** [review]

File: internal/dropdown/src/main/java/com/decathlon/vitamin/dropdown/VitaminDropdownAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            viewHolder = convertView.tag as ViewHolder
+            convertView
+        }
+        viewHolder.bind(getItem(position))
+        return newConvertView
+    }
+
+    override fun getFilter(): Filter = ArrayFilter(items, this)
+
+    private class ArrayFilter<T>(val list: List<T>, val adapter: BaseAdapter) : Filter() {
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply internal/dropdown/src/main/java/com/decathlon/vitamin/dropdown/VitaminDropdownAdapter.kt:43]: Yes, it is required by the `AutocompleteTextView` in a `TextInputLayout`...

File: internal/dropdown/src/main/java/com/decathlon/vitamin/dropdown/VitaminDropdownAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            if (results.count > 0) {
+                adapter.notifyDataSetChanged()
+            } else {
+                adapter.notifyDataSetInvalidated()
+            }
+        }
+    }
+}
+
+private class ViewHolder(view: View) {
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply internal/dropdown/src/main/java/com/decathlon/vitamin/dropdown/VitaminDropdownAdapter.kt:61]: Not at all. Remove both class outside the Adapter!

File: internal/dropdown/src/main/res/layout/vtmn_dropdown_item.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.appcompat.widget.LinearLayoutCompat xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    style="?attr/vtmnDropDownItemStyle"
```
[the remark this answers is not in the document]
Comment (GerardPaligot) [reply internal/dropdown/src/main/res/layout/vtmn_dropdown_item.xml:5]: There is no impact. I use the same style for the simple dropdown item and this one and text attributes is ignored for this one. I'm open to your feedback:

* I create two styles and ask to consumers to declare both styles on their theme if they use only this artifact
* I create a single style for both layout but there is some attributes which is ignored for this one. But for consumers, they need to configure and override only one style too.

I'm not sure what is the best option.

File: text-inputs/README.md [hunk 4]
```diff
@@ -182,4 +248,28 @@ implementation("com.decathlon.vitamin:vitamin:<version>")
     </com.decathlon.vitamin.textinputs.VitaminTextInputLayoutFilled>
 
 </androidx.constraintlayout.widget.ConstraintLayout>
-```
\ No newline at end of file
+```
+
+## Dropdown Text Input
+
+The Vitamin subclass of `MaterialAutoCompleteTextView`, `VitaminDropdownEditText`, declare 3
```
Comment (MaximeJineMaraval) [inline text-inputs/README.md:255]: Should not be "declare 2"?

File: menus/README.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+// Popup
+val popup = PopupMenu(requireContext(), v, Gravity.NO_GRAVITY, R.attr.popupMenuStyle, R.style.Widget_Vitamin_Menu)
+
+// ListPopup
+val listPopup = ListPopupWindow(requireContext(), null, R.attr.listPopupWindowStyle, R.style.Widget_Vitamin_Menu_List)
+```
+
+## Dropdown List Menu
+
+The Vitamin subclass of `ListPopupWindow`, `VitaminListMenu`, declare 3
```
Comment (MaximeJineMaraval) [inline menus/README.md:112]: Should not be "declare 2"?

File: internal/dropdown/src/main/res/layout/vtmn_dropdown_item.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (MaximeJineMaraval) [reply internal/dropdown/src/main/res/layout/vtmn_dropdown_item.xml:5]: Ok, understood. 
I think we could keep this version right now, as it's easier to use for the user. But this should be re-think if we have to add more complexity later.

**Review by MaximeJineMaraval — APPROVED** [review]

**Pull request merged**